lib/common/memory.c

```diff
@@ -413,3 +413,29 @@ void h2o_append_to_null_terminated_list(void ***list, void *element)
     (*list)[cnt++] = element;
     (*list)[cnt] = NULL;
 }
+
```

Comment:
Thank you for the changes. To clarify, my suggestion is to do 
```
#define _POSIX_C_SOURCE 200112L
#undef _GNU_SOURCE
```
here, so that the POSIX variant would be used below.

We can retain `h2o_strerror_r` returning void as-is.

Reply:
So the choice is using POSIX version instead of gnu version.
Should this be defined at the **beginning** of file before any `#include` (or at lease before `#include <string.h>`)?

Reply:
@chenbd I think you are correct. Using macros is not a good way to resolve the issue.

So it would either be calling `__xpg_strerror_r` (it's part of the linux ABI, see http://refspecs.linux-foundation.org/LSB_3.2.0/LSB-Core-generic/LSB-Core-generic/baselib-xpg-strerror_r-3.html), or creating a wrapper for `strerror_r`.

I think that your approach of creating a wrapper is better, and I think the wrapper API we should provide is the one that returns `char *`; it's theoretically faster and it's easier to mimic. WDYT?

Reply:
yes, I prefer `char *h2o_strerror_r(int errnum, char *buf, size_t buflen);`